test/deployframework/framework.go

```diff
@@ -0,0 +1,335 @@
+}
+
+// WaitForMeteringPods periodically polls the list of pods in the @namespace
+// and ensures the metering pods created are considered ready. In order to exit
+// the polling loop, the number of pods listed must match the expected number
+// of @targetPods, and all pod containers listed must report a ready status.
+func (df *DeployFramework) WaitForMeteringPods(targetPods int, namespace string) (bool, error) {
+	var readyPods []string
+	var unreadyPods []podStat
+
+	start := time.Now()
+
+	df.Logger.Infof("Waiting for all metering pods to be ready")
+	err := wait.Poll(10*time.Second, 20*time.Minute, func() (done bool, err error) {
+		unreadyPods = nil
+		readyPods = nil
+
+		pods, err := df.Client.CoreV1().Pods(namespace).List(meta.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		if len(pods.Items) == 0 {
+			return false, fmt.Errorf("the number of pods in the %s namespace should exceed zero", namespace)
+		}
+
+		for _, pod := range pods.Items {
+			podIsReady, readyContainers := df.checkPodStatus(pod)
+			if podIsReady {
+				readyPods = append(readyPods, pod.Name)
+				continue
+			}
+
+			unreadyPods = append(unreadyPods, podStat{
+				PodName: pod.Name,
+				Ready:   readyContainers,
+				Total:   len(pod.Status.ContainerStatuses),
+			})
+		}
+
+		df.logPollingSummary(targetPods, readyPods, unreadyPods)
+
+		return len(pods.Items) == targetPods && len(unreadyPods) == 0, nil
+	})
+	if err != nil {
+		return false, fmt.Errorf("the metering pods failed to report a ready status before the timeout period occurred: %v", err)
+	}
+
+	df.Logger.Infof("Installing metering took %v", time.Since(start))
+
+	return true, nil
+}
+
+// GetRouteBearerToken queries the @namespace for the reporting-operator service account and attempts
+// to find the secret that contains the reporting-operator token. If that secret exists, return the
+// string representation of the token (key) byte slice (value), or return an error.
+func (df *DeployFramework) GetRouteBearerToken(namespace string) (string, error) {
+	var sa *v1.ServiceAccount
+	var err error
+
+	df.Logger.Infof("Waiting for the reporting-operator service account to be created")
+	err = wait.Poll(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+		sa, err = df.Client.CoreV1().ServiceAccounts(namespace).Get(reportingOperatorServiceAccountName, meta.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+
+		df.Logger.Infof("The %s service account has been created", reportingOperatorServiceAccountName)
+		return true, nil
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to get the %s service account before timeout has occurred: %v", reportingOperatorServiceAccountName, err)
+	}
+
+	if len(sa.Secrets) == 0 {
+		return "", fmt.Errorf("failed to return a list of secrets in the reporting-operator service account")
+	}
+
+	var secretName string
+
+	for _, secret := range sa.Secrets {
+		if strings.Contains(secret.Name, "token") {
+			secretName = secret.Name
```

Comment:
`break` from the loop when you find the item you want.

Reply:
whoops, good call